src/qt/guiutil.cpp

```diff
@@ -409,7 +480,19 @@ bool SetStartOnSystemStartup(bool fAutoStart)
             WCHAR pszExePath[MAX_PATH];
             GetModuleFileNameW(NULL, pszExePath, sizeof(pszExePath));
 
-            TCHAR pszArgs[5] = TEXT("-min");
+            std::wstring autostartup_arguments;
+            std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+
+            if (!autostartup.data_dir.empty())
+            {
+                autostartup_arguments = converter.from_bytes("-datadir=")
+                        + autostartup.data_dir.wstring()
+                        + converter.from_bytes(" ");
```

Comment:
```suggestion
                autostartup_arguments = converter.from_bytes("-datadir=\"")
                        + autostartup.data_dir.wstring()
                        + converter.from_bytes("\" ");
```
Need to shell-quote the `-datadir` argument in case that path has spaces in any of the directory names.